src/gui/map/map_find_feature.cpp

```diff
@@ -172,46 +182,41 @@ ObjectQuery MapFindFeature::makeQuery() const
 
 void MapFindFeature::findNext()
 {
-	auto map = controller.getMap();
-	auto first_object = map->getFirstSelectedObject();
-	map->clearObjectSelection(false);
-	
-	Object* next_object = nullptr;
 	auto query = makeQuery();
 	if (!query)
 	{
 		if (auto window = controller.getWindow())
 			window->showStatusBarMessage(OpenOrienteering::TagSelectWidget::tr("Invalid query"), 2000);
 		return;
 	}
+	
+	auto* map = controller.getMap();
+	Object* first_match = nullptr;  // the first match in all objects
+	Object* pivot_object = map->getFirstSelectedObject();
+	Object* next_match = nullptr;   // the next match after pivot_object
+	map->clearObjectSelection(false);
+	
+	auto search = [&first_match, &pivot_object, &next_match](Object* object) {
+		if (next_match)
+			return;
 		
-	auto search = [&first_object, &next_object, &query](Object* object) {
-		if (!next_object)
+		bool after_pivot = (pivot_object == nullptr);
+		if (object == pivot_object)
+			pivot_object = nullptr;
+		
+		if (isSelectable(object))
 		{
-			if (first_object)
-			{
-				if (object == first_object)
-					first_object = nullptr;
-			}
-			else if (query(object))
-			{
-				next_object = object;
-			}
+			if (after_pivot && !next_match)
```

Comment:
```suggestion
			if (after_pivot)
```
Already shortcutting on `next_match` in line 200.